package/MDAnalysis/core/groups.py

```diff
@@ -115,8 +115,8 @@
 from ._get_readers import get_writer_for, get_converter_for
 
 
-def _unpickle(u, ix):
-    return u.atoms[ix]
+def _unpickle(cls, ix, u):
+    return cls(ix, u)
```

Comment:
Does this change affect unpickling of pickles that were created with earlier versions?

(We don't encourage pickling universes to save state but it's allowed. If it breaks old pickles then it would be a *backward-incompatible* change.)

Reply:
It's neat.

Reply:
You're absolutely right. I rewrote it to make it backwards compatible with a warning.
Fixed on #3953.